test/test_nn.py

```diff
@@ -18936,19 +18927,16 @@ def expected_output(dim):
             expected_indices = expected_indices(num_dim)
             expected_output = expected_output(num_dim)
             self.assertEqual(indices.dim(), input.dim())
-            # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-            self.assertEqualIgnoreType(indices.data.squeeze(), expected_indices)
-            # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-            self.assertEqualIgnoreType(output.data.squeeze(), expected_output)
+            self.assertEqual(indices.data.squeeze(), expected_indices.to(dtype=indices.data.squeeze().dtype))
```

Comment:
Probably better change to `self.assertEqual(indices.data.squeeze(), expected_indices, exact_dtype=False)`.
Because converting one operand's dtype to another operand's dtype basically ignores types, but `exact_dtype=False` is shorter and more clear.
Here, similarly next line, and line 18939.